src/components/ProductDetail/ProductDetailOrderForm.js

```diff
@@ -0,0 +1,138 @@
+import { useEffect, useState } from "react";
+import SizeOptionsContainer from "./SizeOptionsContainer";
+import { camelToPascalCase } from "../../helperFunctions";
+import { Link } from "react-router-dom";
+
+const ProductDetailOrderForm = ({product, addToShoppingBag, shoppingBag, updateQuantity}) => {
+  const [isSingleSize, setIsSingleSize] = useState(false);
+  const [inputFields, setInputFields] = useState({
+    "color":"",
+    "size": "",
+    "quantity":0
+  });
+  const [colorOptions, setColorOptions] = useState([]);
+  const [isFormHealthy, setIsFormHealthy] = useState(false);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+  
+  useEffect(() => {
+    setColorOptions(product?.colorOptions)
+  }, [product]);
+  
+  useEffect(() => {
+    checkFormHealth()
+  }, [inputFields]);
+
+  useEffect(() => {
+    console.log(isFormHealthy)
+  }, [isFormHealthy]);
+
+  useEffect(() => {
+    const total = shoppingBag.reduce((total, item) => total + item.quantity, 0);
+    setTotalQuantity(total);
+  }, [shoppingBag])
+
+  const checkFormHealth = () => {
+    if (inputFields.color && inputFields.size && inputFields.quantity) {
+      setIsFormHealthy(true);
+    } else {
+      setIsFormHealthy(false);
+    }
+  }
+
+
+  const handleSelect = (e, changedField) => {
+    const clonedInputs = {...inputFields};
+    clonedInputs[changedField] = e.target.value;
+    setInputFields(clonedInputs);
+    checkFormHealth();
+  }
+
+  const ColorOptionsCode = () => {
+    const options = colorOptions?.map(color => {
+      return <option key={color} value={color}> { camelToPascalCase(color) } </option>
+    })
+    return (
+      <>
+        <option value="">Choose color</option>
+        {options}
+      </>
+    );
+  }
+  
+  const checkForExistingMatch = (cart, item) => {
+    const result = cart.find(existingItem => {
+      return (existingItem.type?.toLowerCase() === item.type?.toLowerCase()) && (existingItem.size === item.size) && (existingItem.color === item.color)
+    });
+    return result?.id;
```

Comment:
what is the ? doing here?

Reply:
In the case that the shopping card does not find a matching item, result would be undefined. And hence, would not have an attribute of `id`. This would lead to an error. 
`result?.id` asks javascript to first evaluate whether result exists (and if it exists, it'd definitely have an attribute of id) so there won't be any error logged.

Reply:
oh that makes sense thanks